src/package/init.ts

```diff
@@ -17,7 +17,11 @@ type VNodeQueue = VNode[]
 const emptyNode = vnode('', {}, [], undefined, undefined)
 
 function sameVnode (vnode1: VNode, vnode2: VNode): boolean {
-  return vnode1.key === vnode2.key && vnode1.sel === vnode2.sel
+  const isSameKey = vnode1.key === vnode2.key
+  const isSameIs = vnode1.data?.is === vnode2.data?.is
+  const isSameSel = vnode1.sel === vnode2.sel
+
+  return isSameSel || isSameKey || isSameIs
```

Comment:
Is this intended? Actually I wanted to write a thunk version that lets me change `sel` on the init hook, so it'd be great if key equality is enough when they're both defined. But I was surprised to find this when checking the current code: using `||` here means two nodes with same selector will be considered the same even if they have different keys, which doesn't look right.

Reply:
No. I think this was a mistake introduced by 39c9da4615d248265ff8de376a95adab799ca981.

Reply:
Thanks to you for adding the feature and the quick reply!